src/ILOManager.sol

```diff
@@ -51,66 +51,87 @@ contract ILOManager is IILOManager, Ownable, Initializable {
+        return _projects[projectId];
     }
 
     /// @inheritdoc IILOManager
-    function initILOPool(InitPoolParams calldata params) external override onlyProjectAdmin(params.uniV3Pool) returns (address iloPoolAddress) {
-        Project storage _project = _cachedProject[params.uniV3Pool];
+    function initILOPool(InitPoolParams calldata params) external override onlyProjectAdmin(params.projectId) returns (address iloPoolAddress) {
+        // dont need to check if project is exist because only project admin can call this function
+        Project storage _project = _projects[params.projectId];
+        _checkTicks(params.tickLower, params.tickUpper, _project.fee);
         {
-            require(_project.uniV3PoolAddress != address(0), "NI");
             // validate time for sale start and end compared to launch time
             require(params.start < params.end && params.end < _project.launchTime, "PT");
             // this salt make sure that pool address can not be represented in any other chains
             bytes32 salt = keccak256(abi.encodePacked(
                 ChainId.get(),
-                params.uniV3Pool,
-                _initializedILOPools[params.uniV3Pool].length
+                params.projectId,
+                _initializedILOPools[params.projectId].length
             ));
             iloPoolAddress = Clones.cloneDeterministic(ILO_POOL_IMPLEMENTATION, salt);
-            emit ILOPoolCreated(_project.uniV3PoolAddress, iloPoolAddress, _initializedILOPools[params.uniV3Pool].length);
+            emit ILOPoolCreated(params.projectId, iloPoolAddress, _initializedILOPools[params.projectId].length);
         }
 
-        uint160 sqrtRatioLowerX96 = TickMath.getSqrtRatioAtTick(params.tickLower);
-        uint160 sqrtRatioUpperX96 = TickMath.getSqrtRatioAtTick(params.tickUpper);
-        require(sqrtRatioLowerX96 < _project.initialPoolPriceX96 && sqrtRatioLowerX96 < sqrtRatioUpperX96, "RANGE");
-
         IILOPool.InitPoolParams memory initParams = IILOPool.InitPoolParams({
-            uniV3Pool: params.uniV3Pool,
+            projectId: params.projectId,
             tickLower: params.tickLower,
             tickUpper: params.tickUpper,
-            sqrtRatioLowerX96: sqrtRatioLowerX96,
-            sqrtRatioUpperX96: sqrtRatioUpperX96,
-            hardCap: params.hardCap,
-            softCap: params.softCap,
-            maxCapPerUser: params.maxCapPerUser,
+            maxRaise: params.maxRaise,
+            minRaise: params.minRaise,
+            maxRaisePerUser: params.maxRaisePerUser,
             start: params.start,
             end: params.end,
             vestingConfigs: params.vestingConfigs
         });
         IILOPool(iloPoolAddress).initialize(initParams);
-        _initializedILOPools[params.uniV3Pool].push(iloPoolAddress);
+        _initializedILOPools[params.projectId].push(iloPoolAddress);
+    }
+
+    function _checkTicks(int24 tickLower, int24 tickUpper, uint256 fee) internal pure {
```

Comment:
Let's comment on the purpose of this function.
Also if fee = 100, no need to validate?

Reply:
Tick must be multiplies of tickSpacing, in case of fee = 100, tickSpacing = 1, so no validation is required

Reply:
Tick must be multiplies of tickSpacing, in case of fee = 100, tickSpacing = 1, so no validation is required